### DISCUSSION TITLE
About Constant Pressure Gradient Strategy for the Channel Flow

### DISCUSSION THREAD

**Discussion by MarcoArtiano** [xcompact3d/Incompact3d#187] [Q&A] [answered]
Hello guys, I'm studying the Channel Flow with Xcompact3D, but I was wondering how the constant pressure gradient has been implemented. In the code its implementation has the following formulation

```
  if (cpg) then
    re_cent = (re/0.116_mytype)**(1.0_mytype/0.88_mytype)
    xnu = one/re_cent ! viscosity based on Re_cent to keep same scaling as CFR
    !
    fcpg = two/yly * (re/re_cent)**2
  end if
```
and I can't understand why, also it shouldn't be equivalent to set something like this ? Following [https://home.aero.polimi.it/quadrio/papers/2015-EJMB.pdf](url)
```
  if (cpg) then
   ! where xnu = re_tau 
    fcpg = 1
  end if
```
Is this related to the initial conditions that should be changed in the second case ?
Thanks for the help.

**Comment by mathrack** [comment, marked as answer]
If the imposed pressure gradient is one, then all quantities should be in wall-units. The current setup has a constant pressure gradient but the velocity is not in wall-units.

**Reply by MarcoArtiano** [reply]
Thank you so much ! Where can I find the theoretical background for such strategy (that relates the Retau to the Reynolds number based on the centerline velocity and how one ended up with such formulas), and how eventually to express the initial condition in wall units ?

**Reply by mathrack** [reply]
Hello,

The correlation probably comes from some papers... but I do not have the reference. If you find the reference, feel free to share it.

I do not really understand your comment about expressing the initial condition in wall units.

Cheers
